src/commands/flow/internal/checkPredecessorDeadlock.ts

```diff
@@ -0,0 +1,31 @@
+import { FlowWave } from '../../../types'
+
+export const checkPredecessorDeadlock = (flows: {
+  [key: string]: FlowWave
+}): string[] | false => {
+  for (const flowName in flows) {
+    const result = checkPredecessorDeadlockRecursive(flows, flowName, [])
+    if (result) return result
+  }
+
+  return false
+}
+
+const checkPredecessorDeadlockRecursive = (
+  flows: { [key: string]: FlowWave },
+  flowName: string,
+  chain: string[]
+): string[] | false => {
```

Comment:
it is not clear what will contain an array of strings, it would be nice to return an object with descriptive keys. it would be also nice to create a type for it

Reply:
provided type for this `PredecessorDeadlockChain`